services/s3/src/main/java/software/amazon/awssdk/services/s3/handlers/EndpointAddressInterceptor.java

```diff
@@ -59,7 +59,7 @@ public SdkHttpFullRequest modifyHttpRequest(Context.ModifyHttpRequest context, E
 
         if (advancedConfiguration == null || !advancedConfiguration.pathStyleAccessEnabled()) {
             sdkRequest.getValueForField("Bucket", String.class).ifPresent(b -> {
-                if (BucketUtils.isValidDnsBucketName(b, false)) {
+                if (BucketUtils.isValidDnsBucketName(b, false) && !b.contains(".")) {
```

Comment:
+1

Reply:
I am not sure where else this method is used but you can consider renaming and inverting it to shouldUsePathStyle or something like that to avoid the 'yes this is a valid DNS name but s3 doesn't think so' situation.

Reply:
Added a new method isVirtualAddressingCompatibleBucketName calls isValidDnsBucketName and checks if bucket name has periods.